src/clj/fluree/db/reasoner.cljc

```diff
@@ -218,24 +235,42 @@
             (recur r db**))
           db*)))))
 
+(defn deduplicate-raw-rules
+  [raw-rules]
+  (let [rule-ids (map first raw-rules)
+        duplicate-ids (filter #(< 1 (last %)) (frequencies rule-ids))]
+    (reduce (fn [rules [duplicate-id occurances]]
+              (let [grouped-rules (group-by #(= duplicate-id (first %)) rules)]
+                (loop [suffix occurances
+                       rules-to-update (get grouped-rules true)
+                       updated-rules-list (get grouped-rules false)]
+                  (if (empty? rules-to-update)
+                    updated-rules-list
+                    (let [updated-rule [(str duplicate-id suffix) (last (first rules-to-update))]]
+                      (recur (dec suffix) (rest rules-to-update) (conj updated-rules-list updated-rule)))))))
+            raw-rules duplicate-ids)))
+
 (defn reason
```

Comment:
I think this adds complexity to the API in the wrong spot.

If working directly with the API, I think one could instead do:
```
(-> db 
    (reason :owl [{...some json-ld-rules}])
    (reason :datalog some-rules-db)
    (reason :owl some-other-rules-db))
```

Supporting multiple at once is more an issue for stateless transactions, and therefore any special syntax for it should be moved to `query-connection` (or successors) where we handle stateless interactions.

What still needs to be built in doing the above is to "merge in" each successive rule set - and whatever logic we decide on for that (per my other comments) should get implemented there.